synapse/account_validity/__init__.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Any, List, Tuple
+
+if TYPE_CHECKING:
+    import synapse.events
+    import synapse.server
+
+
+class AccountValidity:
```

Comment:
I looked at other modules (namely spam checker and third party event rules) which have this code (instantiating + calling hooks on modules) in a dedicated class outside of handlers, and thought of the account validity handler as the legacy code that'll eventually go away.

Reply:
hrm; I'm not convinced that's a great model. 

spam-checker seems to be an entire module to define three constants - I'm not really sure why we did that.

In my mind, the REST layer calls things in the handlers. It might be better to rename the existing `AccountValidityHandler` (to `LegacyAccountValidityImpl` or something) but keep it in the same file. Then you define a new `AccountValidityHandler` which is called by the REST layer, and manages the calls out to the module as well as the fallback to the `LegacyAccountValidityImpl`.